src/Components/Server/src/Microsoft.AspNetCore.Components.Server.csproj

```diff
@@ -20,6 +20,16 @@
     <Reference Include="Microsoft.Extensions.Caching.Memory" />
     <Reference Include="Microsoft.Extensions.FileProviders.Composite" />
     <Reference Include="Microsoft.Extensions.FileProviders.Embedded" />
+
+    <!-- Add a project dependency without reference output assemblies to enforce build order -->
+    <!-- Applying workaround for https://github.com/microsoft/msbuild/issues/2661 and https://github.com/dotnet/sdk/issues/952 -->
+    <ProjectReference
+      Include="..\..\Web.JS\Microsoft.AspNetCore.Components.Web.JS.npmproj"
+      ReferenceOutputAssemblies="false"
+      SkipGetTargetFrameworkProperties="true"
+      UndefineProperties="TargetFramework"
+      Private="false"
+      Condition="'$(BuildNodeJS)' != 'false'" />
```

Comment:
Do we really need this here and in `Directory.Build.props`?

Reply:
The reference in Directory.Build.props is only added if the property `ReferenceBlazorBuildLocally` is set to true. I assumed that property was not set in this project for a reason so I added the reference directly.

Reply:
I think this is only needed as a direct reference of `M.A.Components.Server` and `M.A.Blazor.Build` - those are the projects that directly ship this file. The other projects will end up with a transitive reference to one of those two.

Reply:
The current implementation does not add a direct reference to M.A.Blazor.Build. Instead, a target file is imported directly: https://github.com/aspnet/AspNetCore/blob/9851e0168812471e52c4c965a0c0005240f762ba/src/Components/Directory.Build.targets#L10. I think your proposal works, but I didn't see a need to change the behaviour.